src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Ssl.cs

```diff
@@ -180,16 +172,37 @@ internal static byte[] ConvertAlpnProtocolListToByteArray(List<SslApplicationPro
                 protocolSize += protocol.Protocol.Length + 1;
             }
 
-            byte[] buffer = new byte[protocolSize];
+            return protocolSize;
+        }
+
+        internal static void SerializeAlpnProtocolList(List<SslApplicationProtocol> applicationProtocols, Span<byte> buffer)
+        {
+            Debug.Assert(GetAlpnProtocolListSerializedLength(applicationProtocols) == buffer.Length,
+                "GetAlpnProtocolListSerializedSize(applicationProtocols) == buffer.Length");
+
             var offset = 0;
             foreach (SslApplicationProtocol protocol in applicationProtocols)
             {
-                buffer[offset++] = (byte)(protocol.Protocol.Length);
-                protocol.Protocol.Span.CopyTo(buffer.AsSpan(offset));
+                buffer[offset++] = (byte)protocol.Protocol.Length;
+                protocol.Protocol.Span.CopyTo(buffer.Slice(offset));
                 offset += protocol.Protocol.Length;
             }
+        }
+
+        internal static unsafe int SslSetAlpnProtos(SafeSslHandle ssl, List<SslApplicationProtocol> applicationProtocols)
+        {
+            int length = GetAlpnProtocolListSerializedLength(applicationProtocols);
+            Span<byte> buffer = length <= 256 ? stackalloc byte[length] : new byte[length];
```

Comment:
```suggestion
            Span<byte> buffer = length <= 256 ? stackalloc byte[256] : new byte[length];
```

Reply:
👍 This would address my concern.

Reply:
I have no problem with the change, but can I ask why, @stephentoub? Is there other reason besides the overflow problem mentioned by @vcsjones?

Reply:
> Is there other reason besides the overflow problem

`stackalloc` of a constant size is more efficient than if you pass it a variable.
*provided you are compiling with `[SkipLocalsInit]`, which runtime is.

Reply:
It makes it completely unambiguous from a security perspective how much is getting allocated on the stack.